examples/mfa/webauthn/server.go

```diff
@@ -0,0 +1,204 @@
+package webauthn
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/go-webauthn/webauthn/webauthn"
+	"github.com/openpubkey/openpubkey/client/cosigner/mfa"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+)
+
+type Server struct {
+	uri      string
+	doneChan chan error
+	user     User
+	auth     *webauthn.WebAuthn
+	session  *webauthn.SessionData
+}
+
+type User interface {
+	webauthn.User
+	AddCredential(cred webauthn.Credential)
+}
+
+var _ mfa.Authenticator = (*Server)(nil)
+
+func New() (*Server, error) {
+	server := &Server{
+		doneChan: make(chan error),
+	}
+
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, fmt.Errorf("failed to bind to an available port: %w", err)
+	}
+	server.uri = fmt.Sprintf("http://localhost:%d", listener.Addr().(*net.TCPAddr).Port)
+
+	// WebAuthn configuration
+	cfg := &webauthn.Config{
+		RPDisplayName: "OpenPubkey",
+		RPID:          "localhost",
+		RPOrigin:      server.uri,
+	}
+
+	server.auth, err = webauthn.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	http.Handle("/", http.FileServer(http.Dir("webauthn/static")))
+
+	http.HandleFunc("/check-registration", server.checkIfRegistered)
+
+	http.HandleFunc("/register/begin", server.beginRegistration)
```

Comment:
This is a non-blocker for this PR and example, but I created an issue to think about how we prevent CSRF or if this is safe. 
https://github.com/openpubkey/openpubkey/issues/57

Resolve after reading